### ISSUE TITLE
[BUG] Custom rule not triggered for tackle-testapp

### ISSUE THREAD

**Issue by sshveta** [konveyor/analyzer-lsp#1095]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

upstream

### Priority

Critical

### Current Behavior

for [tackle-testapp](https://github.com/konveyor/tackle-testapp) the dependency is present in [pom.xml](https://github.com/konveyor/tackle-testapp/blob/110b29a82bec8836216f715850438aa24a2fd2de/pom.xml#L65).

Custom rule tested is 
================
```
- category: potential
  customVariables: []
  description: CUSTOM RULE FOR DEPENDENCIES | jackson-databind identified in the pom.xml
  effort: 1
  labels:
    - konveyor.io/include=always
    - tag
  links:
    - title: Test-002 Test link
      url: http://nufc.com
  message: Test-002 Custom rule
  ruleID: Test-002-00001
  when:
    java.dependency:
      lowerbound: 0.0.0
      name: com.fasterxml.jackson.core.jackson-databind
```

### How Reproducible

Always (Default)

### Steps To Reproduce

- Analysis mode: "Source code + dependencies"
  - Source: https://github.com/konveyor/tackle-testapp
  - Maven credentials: Required and provided
  - Open source libraries: true

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by eemcmullan** [comment]
I am not able to reproduce using latest analyzer and Java provider. Both of these custom rules create violations. Do you mind providing some logs?
